### PR TITLE
1.2.0 - executions pause and resume

### PR DESCRIPTION
**Pull request by igcherkaev**
## 1.2.0

- Added `st2.executions.pause` and `st2.executions.resume` actions.

### CODE REVIEW

File: actions/executions_pause.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class St2ExecutionsPause(St2BaseAction):
+    def run(self, ids):
+        result = {}
+        for i in ids:
+            try:
+                res = self.client.liveactions.pause(execution_id=i)
+            except Exception as exc:
+                result[i] = '{}'.format(exc)
```
Comment (nmaludy) [inline actions/executions_pause.py]:
```suggestion
                result[i] = str(exc)
```

File: actions/executions_pause.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            else:
```
Comment (nmaludy) [inline actions/executions_pause.py:19]: Also, if one of these fails (exception) then we should probably stop and fail the action, right now the action will always succeed.

File: actions/executions_resume.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from lib.action import St2BaseAction
+
+__all__ = [
+    'St2ExecutionsResume'
+]
+
+
+def format_result(item):
```
Comment (nmaludy) [inline actions/executions_resume.py]: this function looks identical to the one in the file above, it should go into the base action so code is not copy/pasted.

File: actions/executions_resume.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    if not item:
+        return None
+
+    return item.to_dict()
```
Comment (nmaludy) [inline actions/executions_resume.py]:
```suggestion
    return item.to_dict() if item else None
```

**Review by nmaludy — CHANGES_REQUESTED** [review]

**Comment by emptywee** [comment]
Yup, and I'll add one more action - `executions.cancel` to have full hand of actions to manage executions :)

File: actions/executions_pause.py (hunk 2 above)
Comment (emptywee) [reply actions/executions_pause.py:19]: yes, it's coming! Realized that after running some real-world scenarios :)

File: actions/executions_pause.py (hunk 1 above)
Comment (emptywee) [reply actions/executions_pause.py]: You got it!

File: actions/executions_resume.py (hunk 3 above)
Comment (emptywee) [reply actions/executions_resume.py]: :) agreed

File: actions/executions_resume.py (hunk 4 above)
Comment (emptywee) [reply actions/executions_resume.py]: ok!

File: actions/executions_pause.py (hunk 2 above)
Comment (emptywee) [reply actions/executions_pause.py:19]: but not sure if we need to stop? I thought of returning overall result True/False and a dict of results. Perhaps, we can add a flag to ignore/stop on errors as action's parameter? Maybe in a future release if such feature is requested?

File: actions/executions_cancel.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class St2ExecutionsCancel(St2BaseAction):
+    def run(self, ids):
+        result = {}
+        success = True
+        for i in ids:
+            try:
+                res = self.client.executions.delete_by_id(instance_id=i)
+                if 'faultstring' in res and res['faultstring']:
```
Comment (nmaludy) [inline actions/executions_cancel.py]: This could be simplified to 
```suggestion
                if res.get('faultstring'):
```

**Review by nmaludy — CHANGES_REQUESTED** [review]

File: actions/executions_cancel.py (hunk 5 above)
Comment (emptywee) [reply actions/executions_cancel.py]: fixed, in the upcoming PR

**Review by nmaludy — APPROVED** [review]

**Pull request merged**